### ISSUE TITLE
[CI] ImportError: cannot import name 'ThreadLocalMetaData' from 'sqlalchemy.schema' (/usr/local/lib/python3.8/dist-packages/sqlalchemy/schema.py) 

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#481]
For the record, the import error mentioned in #477 is sill sometimes appearing (see https://github.com/fractal-analytics-platform/fractal-server/actions/runs/4024017688/jobs/6915556627), and it looks like
```python traceback
STDERR:
Content of /tmp/pytest-of-runner/pytest-0/test_failing_workflow_JobExecu0/artifacts-test_failing_workflow_JobExecutionError/workflow_000001_job_000001/0.slurm.9.err:
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 185, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.8/runpy.py", line 111, in _get_module_details
    __import__(pkg_name)
  File "/usr/local/lib/python3.8/dist-packages/fractal_server/app/runner/__init__.py", line 29, in <module>
    from ..models import ApplyWorkflow
  File "/usr/local/lib/python3.8/dist-packages/fractal_server/app/models/__init__.py", line 4, in <module>
    from ...common.schemas import *  # noqa F401
  File "/usr/local/lib/python3.8/dist-packages/fractal_server/common/schemas/__init__.py", line 5, in <module>
    from sqlmodel import SQLModel
  File "/usr/local/lib/python3.8/dist-packages/sqlmodel/__init__.py", line 24, in <module>
    from sqlalchemy.schema import ThreadLocalMetaData as ThreadLocalMetaData
ImportError: cannot import name 'ThreadLocalMetaData' from 'sqlalchemy.schema' (/usr/local/lib/python3.8/dist-packages/sqlalchemy/schema.py)
```

**Comment by tcompa** [comment]
First guess: there is a python-path mismatch somewhere.
Note that paths like `/usr/local/lib/python3.8/dist-packages/sqlalchemy/schema.py` are the ones on the slurm container, while the python interpreter running pytest is sitting somewhere else (in a poetry-generated venv) and it's not even necessarily 3.8.

**Comment by tcompa** [comment]
For the record, trying to run everything by hand from within the container doesn't raise the same error:
```
$ docker ps
CONTAINER ID   IMAGE                                     COMMAND                  CREATED         STATUS         PORTS                                                                                NAMES
f09c260bc78d   slurm_docker_images_slurmnode1            "/etc/slurm-llnl/doc…"   9 minutes ago   Up 9 minutes   6817-6819/tcp                                                                        slurm_docker_images_slurmnode1_1
c7a065d75303   slurm_docker_images_slurm-docker-master   "/etc/slurm-llnl/doc…"   9 minutes ago   Up 9 minutes   3306/tcp, 50097/tcp, 0.0.0.0:6817-6819->6817-6819/tcp, :::6817-6819->6817-6819/tcp   slurm_docker_images_slurm-docker-master_1

$ docker exec -it f09c260bc78d bash

admin@slurmnode1:/$ cd

admin@slurmnode1:~$ /usr/bin/python3.8
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> import cloudpickle 
>>> def fun(): return 42
... 
>>> funcser = cloudpickle.dumps((fun, [], {}))
>>> with open("x.in.pickle", "wb") as f:
...     f.write(funcser)
... 
428
>>> 
admin@slurmnode1:~$ /usr/bin/python3.8 /usr/lib/python3.8/runpy.py fractal_server.app.runner._slurm.remote x.in.pickle
admin@slurmnode1:~$ ls
x.in.pickle  x.out.pickle

admin@slurmnode1:~$ /usr/bin/python3.8
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> import cloudpickle
>>> with open("x.out.pickle", "rb") as f: data = f.read()
... 
>>> data
b'\x80\x05\x95\x06\x00\x00\x00\x00\x00\x00\x00\x88K*\x86\x94.'
>>> cloudpickle.loads(data)
(True, 42)
```

**Comment by tcompa** [comment]
Also for the record, `/usr/local/lib/python3.8/dist-packages/sqlalchemy/schema.py` includes the correct line `from .sql.schema import ThreadLocalMetaData  # noqa`.

**Comment by tcompa** [comment]
Last two comments were verified on the slurm node. What about the slurm master node?

```
$ docker ps
CONTAINER ID   IMAGE                                     COMMAND                  CREATED          STATUS          PORTS                                                                                NAMES
f09c260bc78d   slurm_docker_images_slurmnode1            "/etc/slurm-llnl/doc…"   15 minutes ago   Up 15 minutes   6817-6819/tcp                                                                        slurm_docker_images_slurmnode1_1
c7a065d75303   slurm_docker_images_slurm-docker-master   "/etc/slurm-llnl/doc…"   15 minutes ago   Up 15 minutes   3306/tcp, 50097/tcp, 0.0.0.0:6817-6819->6817-6819/tcp, :::6817-6819->6817-6819/tcp   slurm_docker_images_slurm-docker-master_1

$ docker exec -it c7a065d75303 bash

admin@slurm-docker-master:~$ ls /usr/local/lib/python3.8/dist-packages/                    
__pycache__  asttokens  asttokens-2.2.1.dist-info  devtools  devtools-0.10.0.dist-info  executing  executing-1.2.0.dist-info  mpi4py  mpi4py-3.0.3.dist-info  six-1.16.0.dist-info  six.py
```

Question: do we need fractal-server installed on the docker-slurm login node?

**Comment by tcompa** [comment]
Ref: abe4309bcc21d4da392fbcf412ce1a27b6e4c3e2

**Comment by tcompa** [comment]
Quoting from #480:
> We found a major issue in the CI (thanks @ychiucco!), i.e. fractal-server install in docker container was pulling the fresh sqlalchemy 2.0 version (released yesterday), which introduces major changes and is not (yet) supported by our dependencies.
> 
> This is currently addressed by pinning to >=1.4,<2, and we should also come back to this later if a PR like https://github.com/fastapi-users/fastapi-users-db-sqlmodel/pull/13 is merged (I think this is the dependency which is currently holding us back at 1.4.35), to make sure that we would fetch more recent 1.4 versions if possible.
> 
> Meanwhile, another small issue (`cfut_finished_jobs` fixture missing in some tests, and its actual code being slightly different from the cfut=0.5 final version) surfaced, and is fixed here.

Closing this issue, let's open more specific ones if needed.
